packages/zudoku/src/lib/plugins/openapi/playground/QueryParams.tsx

```diff
@@ -82,19 +90,67 @@ export const QueryParams = ({
                   <div className="flex justify-between items-center">
                     <Controller
                       control={control}
-                      render={({ field }) => (
-                        <Input
-                          {...field}
-                          onChange={(e) => {
-                            field.onChange(e.target.value);
-                            if (e.target.value.length > 0) {
-                              form.setValue(`queryParams.${i}.active`, true);
-                            }
-                          }}
-                          placeholder="Enter value"
-                          className="w-full border-0 shadow-none text-xs font-mono"
-                        />
-                      )}
+                      render={({ field }) => {
+                        const hasEnum =
+                          queryParams[i].enum && queryParams[i].enum.length > 0;
+                        const typeOfField = queryParams[i].type;
+                        return hasEnum ? (
+                          typeOfField === "array" ? (
+                            <MultiSelect
+                              options={queryParams[i].enum ?? []}
+                              onValueChange={(value) => {
+                                if (
+                                  Array.isArray(value) &&
+                                  value.length === 0
+                                ) {
+                                  field.onChange("");
+                                } else {
+                                  field.onChange(value);
+                                }
+                              }}
+                              placeholder="Choose options"
+                              value={
+                                Array.isArray(field.value)
+                                  ? field.value.join(",")
+                                  : field.value
+                              }
+                            />
+                          ) : (
+                            <Select
+                              onValueChange={(value) => field.onChange(value)}
+                              value={field.value as string}
+                              defaultValue={field.value as string}
+                            >
+                              <SelectTrigger
+                                className="w-full flex"
+                                placeholder="Choose an option"
+                                value={field.value}
+                              >
+                                <SelectValue />
+                              </SelectTrigger>
```

Comment:
Let's just use `placeholder` that `SelectValue` provides. Then we don't need to add this prop to `SelectTrigger`
So:

```suggestion
                              <SelectTrigger
                                className="w-full flex"
                                value={field.value}
                              >
                                <SelectValue placeholder="Choose an option" />
                              </SelectTrigger>
```